lib/stdlib/test/edlin_expand_SUITE.erl

```diff
@@ -77,16 +84,119 @@ normal(Config) when is_list(Config) ->
       {"module_info",0},
       {"module_info",1}]} = do_expand("expand_test:"),
     {yes,[],[{"a_fun_name",1},
-	     {"a_less_fun_name",1}]} = do_expand("expand_test:a_"),
-    {yes,"arity_entirely()",[]} = do_expand("expand_test:expand0"),
+         {"a_less_fun_name",1}]} = do_expand("expand_test:a_"),
+    {yes,"arity_entirely()",[{"expand0arity_entirely",0}]} = do_expand("expand_test:expand0"),
+    ok.
+
+to_atom(Str) ->
+    case erl_scan:string(Str) of
+    {ok, [{atom,_,A}], _} ->
+        {ok, A};
+    _ ->
+        error
+    end.
+
+type_completion(_Config) ->
+    %% test that edlin_expand:type_traverser works for all preloaded modules in the shell
+    ct:timetrap({minutes, 10}),
+    LoadedModules = [to_atom(M) || {M,_,_} <- code:all_available()],
+    ModExports = [{Mod, edlin_expand:get_exports(Mod)} || {ok, Mod} <- LoadedModules],
+        [[begin
+            Str = atom_to_list(Mod) ++ ":" ++ atom_to_list(Func) ++ "(",
+            io:format("~p~n", [Str]),
+            do_expand(Str)
+          end || {Func, _}<- Exports] || {Mod, Exports} <- ModExports],
```

Comment:
Should the test also include all arguments to each function? That is something like this:

```suggestion
        [lists:foreach(fun({Func, Arity}) ->
          lists:foreach(fun(A) ->
            Str = io_lib:write_atom(Mod) ++ ":" ++ io_lib:write_atom(Func) ++ "(" ++ lists:duplicate(A, "ok,"),
            io:format("~p~n", [Str]),
            do_expand(Str)
          end, lists:seq(0, Arity))
         end, Exports) || {Mod, Exports} <- ModExports],
```

Reply:
The actual type traversal happens when we complete the first parameter, so all errors would already appear there, checking the other parameters dont give us much I think, if we would traverse down in nestings, that could give us something but then we would need to detect the type of each parameter to see if it takes tuples/maps/lists and add {,#{,[. This will probably take a lot of time though, maybe the function can be optimized further but the test already takes more than 10minutes locally.